csrc/kernel_cache.h

```diff
@@ -222,7 +222,8 @@ class FusionKernelRuntime {
   using HeuristicsPtr = std::unique_ptr<FusionHeuristics>;
   std::optional<HeuristicsPtr> getMaybeHeuristicsFor(
       const KernelArgumentHolder& args,
-      std::optional<PrimDataType> forced_index_type = std::nullopt);
+      std::optional<PrimDataType> forced_index_type = std::nullopt,
+      bool initial_heuristics = false);
```

Comment:
It would be even better if getMaybeHeuristicsFor can tell by itself whether it's the initial attempt. This might make the internal logic a little complicated but get rid of an external knob, which sounds like a win.

Reply:
We determine `initial_heuristics=True` when `heuristics_ == nullptr`. The argument was removed.